R/download_zenodo.R

```diff
@@ -14,30 +15,32 @@
 #' @importFrom jsonlite fromJSON
 #' @importFrom tools md5sum
 #' @importFrom utils tail
-#'
+#' @importFrom assertthat
+#' assert_that
+#' is.string
 #'
 #' @export
 #' @family download
 #'
 #' @examples
 #' \dontrun{
-#' # Single zip file deposition
-#' download_zenodo(".", "10.5281/zenodo.1283345")
-#' # Multiple files deposition
-#' download_zenodo(".", "10.5281/zenodo.1172801")
-#' # Single pdf file deposition
-#' download_zenodo(".", "10.5281/zenodo.168478")
+#' # Single zip file download
+#' download_zenodo(doi = "10.5281/zenodo.1283345")
+#' # Multiple files download
+#' download_zenodo(doi = "10.5281/zenodo.1172801")
+#' # Single pdf file download
+#' download_zenodo(doi = "10.5281/zenodo.168478")
 #' }
-download_zenodo <- function(path, doi) {
-  if (missing(path)) {
-    stop("Please provide a path to which the data need to be downloaded")
-  }
+download_zenodo <- function(doi,
+                            path = ".") {
   if (missing(doi)) {
     stop(paste0("Please provide a doi for a Zenodo archive. This is a ",
                 "string starting with '10.5281/zenodo.' followed by a ",
                 "unique number."))
   }
 
+  assert_that(is.string(doi), is.string(path))
+
   # check for existence of the folder
   if (!dir.exists(path)) {
```

Comment:
You can replace the entire if construction with
```suggestion
stopifnot(dir.exists(path))
```

Reply:
just a note about the suggested commit here, it seems that more will need to be replaced than just line 45

Reply:
Resolved in 4d47aab